modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/utils/DataDrivenTestUtils.java

```diff
@@ -0,0 +1,260 @@
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+package org.wso2.am.integration.tests.restapi.utils;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Cookie;
+import javax.ws.rs.core.Form;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Map;
+
+import org.codehaus.jackson.jaxrs.JacksonJsonProvider;
+import org.wso2.am.integration.tests.restapi.RESTAPITestConstants;
+
+/**
+ * This class is used to send respective requests based on the request
+ */
+public class DataDrivenTestUtils {
+
+
+    private void setKeysForRestClient() {
+        System.setProperty("javax.net.ssl.trustStore", "/repository/resources/security/wso2carbon.jks");
+        System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
+        System.setProperty("javax.net.ssl.trustStoreType", "JKS");
+
+    }
+
+    /**
+     * This method is used to send the request to the REST API based on its type
+     *
+     * @param method http method to be used
+     * @param resourceUrl url of the resource
+     * @param queryParameters map containing query parameters
+     * @param requestHeaders map containing request headers
+     * @param requestPayload request payload/body
+     * @param cookie cookie string if any
+     * @return response of the relevant request
+     */
+    public Response sendRequestToRESTAPI(String method, String resourceUrl, Map<String, String> queryParameters,
+                                         Map<String, String> requestHeaders, String requestPayload, String cookie) {
+
+        Response response = null;
+
+        if (RESTAPITestConstants.GET_METHOD.equalsIgnoreCase(method)) {
+            response = geneticRestRequestGet(resourceUrl, RESTAPITestConstants.APPLICATION_JSON_CONTENT,
+                    queryParameters, requestHeaders, cookie);
+        } else if (RESTAPITestConstants.POST_METHOD.equalsIgnoreCase(method)) {
+            response = geneticRestRequestPost(resourceUrl, RESTAPITestConstants.APPLICATION_JSON_CONTENT,
+                    RESTAPITestConstants.APPLICATION_JSON_CONTENT, requestPayload, queryParameters, requestHeaders,
+                    cookie);
+        } else if (RESTAPITestConstants.PUT_METHOD.equalsIgnoreCase(method)) {
+            response = geneticRestRequestPut(resourceUrl, RESTAPITestConstants.APPLICATION_JSON_CONTENT,
+                    RESTAPITestConstants.APPLICATION_JSON_CONTENT, requestPayload, queryParameters, requestHeaders,
+                    cookie);
+        } else if (RESTAPITestConstants.DELETE_METHOD.equalsIgnoreCase(method)) {
+            response = geneticRestRequestDelete(resourceUrl, RESTAPITestConstants.APPLICATION_JSON_CONTENT,
+                    queryParameters, requestHeaders, cookie);
+        }
```

Comment:
Currently the APIM ~ REST API is only using the methods given in this class. We had a discussion with the team and decided to include only the needed methods for the time being (i.e for APIM 1.10.0 release).  

We hope to include the other http methods when the REST API starts using them in the future. We have also created a JIRA [1] to keep track on this.  

[1] https://wso2.org/jira/browse/APIMANAGER-4415